Kas 2556 pub from mr  link doc to pub

mdv-professional opened the pull request:
[Jira](https://kanselarij.atlassian.net/browse/KAS-2562)

**Opmerkingen**
1. Soms incorrecte resultaten:
    vb. lijst identifications zonder filter:
    - 5 pub5
    - 6 pub6
    - 7
    - 8 pub8
    
    filter "6": 3 resultaten.
    filter "8": 0 resultaten.
    
    Ik vermoed dat het in mu-cl-resource zit, want in mijn code leek het OK.
2. Ik vind de UI-flow niet ideaal, noch in code noch in gebruik:
    Ik ben gebruiker en wil een document van een bestaand naar een nieuwe publicatie flow verplaatsen: Ik moet dan eerst "Geen kiezen"
3. Ik zou het aantal calls kunnen beperken door caching (vb. van initiele lijst). Dit moet dan in de BatchDocumentPublicationModal.

Commit:
KAS-2556 (rename) batch documents publication modal | link doc to pub [pub from mr]

Commit:
KAS-2556 (busy) | link doc to pub [pub from mr]

Commit:
KAS 2556 (busy) | refdoc existing pub [pub from mr]

Commit:
KAS 2556 (busy) | refdoc existing pub [pub from mr]

Commit:
Merge branch 'KAS-2576-pub-from-mr--short-and-long-title--always-set' into KAS-2556-pub-from-mr--link-doc-to-pub

Commit:
KAS 2556 (busy) link selector  | refdoc existing pub [pub from mr]

Commit:
KAS 2556 (busy) show new pub  | refdoc existing pub [pub from mr]

Commit:
KAS 2556 (done) show new pub  | refdoc existing pub [pub from mr]

Commit:
KAS 2556 (done) move  batch-documents-publication-modal | refdoc existing pub [pub from mr]

Commit:
KAS 2556 (fix) restore store | refdoc existing pub [pub from mr]

Commit:
KAS 2556 cleanup + comment | refdoc existing pub [pub from mr]

Commit:
KAS 2556 little improvements | refdoc existing pub [pub from mr]

erikap commented:
@mdv-professional Kan je deze branch nog up-to-date brengen met development? Dat maakt de review iets makkelijker.

erikap committed:
Merge branch 'development' into KAS-2556-pub-from-mr--link-doc-to-pub

erikap committed:
Remove unused argument from publication-flow-selector component

erikap committed:
Update publication-flow-selector component class name

erikap committed:
Configure publication-flow-selector with Dutch messages

erikap committed:
Use debounced search in publication-flow-selector

erikap committed:
Refactor batch-documents-publication-row to a seperate component

Since table cells on one row are dependent on each other, it's easier
to generate a seperate component for each row and handle the shared
state there.

erikap committed:
Limit search on existing publications to pubs linked to same case

erikap commented:
Over je opmerkingen:

1. Had je een specifieke reden om de `:gte:` flag te gebruiken bij het filteren? De identificator is een string, omdat ze ook niet-numerieke characters kan bevatten (vb. 'bis'). Filteren met de `:gte:` flag is voor numerieke waarden. Het gedrag van `:gte:` met string values is ongespecifieerd, dus vandaar de vreemde zoekresultaten, vermoed ik.

2. Aan de UI-flow hebben we weinig te beslissen, vrees ik. Ik heb de code wel wat herwerkt. Aangezien de cellen op 1 rij afhankelijk zijn van elkaar, leek het me beter om deze in een aparte component te steken. Op die manier kan de gedeelde state beheerd worden door die aparte component.

3. Ik heb het aantal calls bij het zoeken wat beperkt door gebruik te maken van een debounced search. Er zijn nog verdere optimalisaties mogelijk om het aantal network requests te beperken, maar dat lijkt mij niet de hoogtse prioriteit nu.

Wat niet in de Jira story gespecifieerd stond, maar ik nog toegevoegd heb: het zoeken naar een bestaande publicatie om een document aan te linken is nu beperkt tot publicaties die reeds gelinkt zijn aan hetzelfde dossier. Anders kunnen we in een toestand komen dat stukken uit verschillende dossiers gelinkt zijn als referentiedocument van een publicatie, terwijl de publicatie maar gelinkt is aan één van die dossiers.

erikap committed:
Inline values of link mode options array

ghost commented:
1. OK. Ik dacht een truk gevonden te hebben op het gebrek aan zoekfilter "startsWith"
2. Ik heb dit niet gedaan omdat ze de rows en de modal aan elkaar gelinkt zijn (new publication modal) en te weinig eigen state hebben (enkel de "linkMode"). Ook is de modal beperkt qua code omvang.

- Ik had ook de EmberPowerSelect iets generieker proberen te maken (1 filter trigger) en de PublicationFlowSelector weggedaan.

erikap committed:
Merge branch 'development' into KAS-2556-pub-from-mr--link-doc-to-pub

erikap committed:
Merge branch 'development' into KAS-2556-pub-from-mr--link-doc-to-pub

erikap committed:
Merge branch 'development' into KAS-2556-pub-from-mr--link-doc-to-pub
